smart-contract/program/src/processor/activate_stake_pool.rs

```diff
@@ -72,6 +67,12 @@ pub fn process_activate_stake_pool(program_id: &Pubkey, accounts: &[AccountInfo]
 
     stake_pool.header.tag = Tag::StakePool as u8;
     stake_pool.header.last_claimed_offset = central_state.last_snapshot_offset;
+    if central_state.last_snapshot_offset > u16::MAX as u64 {
+        return Err(AccessError::Overflow.into());
+    }
+    stake_pool.header.current_day_idx = central_state.last_snapshot_offset as u16;
```

Comment:
This would prevent the pool owner from reaping the rewards before being the pool activated, correct?

Reply:
They wouldn't be able to reap the rewards even with the old implementation as there are none and all the operations that change the total staked in the pool depend on the pool being cranked at first.

The thing that this fixes is that the pool doesn't have to be cranked after activation and before the first use